### ISSUE TITLE
decrbyfloat decreases twice if atomic block returns nil or raises error

### ISSUE THREAD

**Issue by samoliuk** [nateware/redis-objects#249]
```
c = Redis::Counter.new('aaa', start: 1000)

# decrements by 2 if failed
c.decrbyfloat do |v|
  nil
end
=> nil

c.value
=> 998

# decrements by 1 if successful
c.decrbyfloat do |v|
  true
end
=> true

c.value
=> 997
```

Looks like positive `by` should be passed there, not negative, i.e. `-` should not be applied?
https://github.com/nateware/redis-objects/blob/master/lib/redis/counter.rb#L105

**Comment by nateware** [comment]
You're right, this seems like a significant bug and I think you have the right fix.  Could you patch it and submit a pull request with tests?  That would be a HUGE help! CC @tmsrjs

**Comment by samoliuk** [comment]
@nateware I've prepared a pr, but unable to push it
```
$ git push -u origin fix-issue-249-decrbyfloat-with-atomic-block
ERROR: Permission to nateware/redis-objects.git denied to samoliuk.
fatal: Could not read from remote repository.
```

Could you advice how I can push my feature branch?

**Comment by tmsrjs** [comment]
@samoliuk you should push to your own fork of this repo and create a PR from there.

**Comment by samoliuk** [comment]
https://github.com/nateware/redis-objects/pull/251
@tmsrjs @nateware

**Comment by tmsrjs** [comment]
fixed by #251
